### ISSUE TITLE
gluster-deploy design issues

### ISSUE THREAD

**Issue by nandajavarma** [nandajavarma/gluster-deploy#1]
#### Options
- -c conf file
- -h help
- -k, --keep (Do not delete the generated yaml and hostvars)
- -d --dryrun (But dry run what?)
#### Configuration file
- -- Existing config for backend/playbookgen is good.
- GlusterFS
  - -- hosts (We already have)
  - -- Volume name
  - -- Vol type (Replicate, distribute, ... )
  - -- replica count (If vol type is replicate)
  - -- Options (read-ahead off, write-behind on) ...
  - -- Client ip/hostname (If more than one client mount them all)
  - -- Client mount-points
  - -- Samba and NFS-Ganesha (Future)
#### Refactoring playbookgen (Will it continue to be standalone program?)
- Make them libraries.
- Write libraries for all the GlusterFS operations.
  - For example import peer, peer.probe(), peer.detach() ... volume.create, volume.delete?
  - Evaluate if we can reuse Bala's python bindings for GlusterFS?
- Proper documentation as we go.
